core/src/test/scala/unit/kafka/server/ControllerApisTest.scala

```diff
@@ -730,8 +729,45 @@ class ControllerApisTest {
     request.topics().add(new CreatePartitionsTopic().setName("bar").setAssignments(null).setCount(5))
     request.topics().add(new CreatePartitionsTopic().setName("baz").setAssignments(null).setCount(5))
     assertEquals(Set(new CreatePartitionsTopicResult().setName("foo").
-        setErrorCode(NONE.code()).
-        setErrorMessage(null),
+      setErrorCode(NONE.code()).
+      setErrorMessage(null),
+      new CreatePartitionsTopicResult().setName("bar").
+        setErrorCode(INVALID_REQUEST.code()).
+        setErrorMessage("Duplicate topic name."),
+      new CreatePartitionsTopicResult().setName("baz").
+        setErrorCode(TOPIC_AUTHORIZATION_FAILED.code()).
+        setErrorMessage(null)),
+      controllerApis.createPartitions(ANONYMOUS_CONTEXT, request,
+        _ => Set("foo", "bar")).get().asScala.toSet)
+  }
+
+  @Test
+  def testValidateOnlyCreatePartitionsRequest(): Unit = {
```

Comment:
Is this test basically the same as `testCreatePartitionsRequest`? Maybe we can get rid of one and turn the other into a `@ParameterizedTest` with `validateOnly` as the parameter?